Explorer/Assets/DCL/SDKComponents/MediaStream/Systems/VideoEventsSystem.cs

```diff
@@ -70,17 +74,30 @@ private void AppendMessage(in CRDTEntity sdkEntity, in MediaPlayerComponent medi
             );
         }
 
-        private static VideoState GetCurrentVideoState(IMediaControl mediaPlayerControl)
+        private static VideoState GetCurrentVideoState(IMediaControl mediaPlayerControl, double previousPlayingTimeCheck, float lastStateChangeTime)
         {
-            if (mediaPlayerControl.IsPlaying()) return VideoState.VsPlaying;
-            if (mediaPlayerControl.IsPaused()) return VideoState.VsPaused;
-            if (mediaPlayerControl.IsFinished()) return VideoState.VsNone;
-            if (mediaPlayerControl.IsBuffering()) return VideoState.VsBuffering;
-            if (mediaPlayerControl.IsSeeking()) return VideoState.VsSeeking;
+            // Important: while PLAYING or PAUSED, MediaPlayerControl may also be BUFFERING and/or SEEKING.
 
+            if (mediaPlayerControl.IsFinished()) return VideoState.VsNone;
             if (mediaPlayerControl.GetLastError() != ErrorCode.None) return VideoState.VsError;
 
-            return VideoState.VsNone;
+            VideoState state = mediaPlayerControl.IsPaused() ? VideoState.VsPaused : VideoState.VsNone;
```

Comment:
should we return here? or can a video be Paused AND Playing at the same time?

Reply:
uhmmm makes sense, tackled at https://github.com/decentraland/unity-explorer/pull/2474/commits/ccd49e4ca406760e9fc5ce20c3bf4f4edd8bf2f0